### DISCUSSION TITLE
1.37.0

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#22379] [General]
Feel free to discuss the [v1.37.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.37.0) release here!

**Comment by sam3487** [comment]
With the new ember settings, do we still need to be running the edge version of z2m or is the normal version fine to use now? Thanks!

**Reply by Nerivec** [reply]
`ember` has been available in release version since [1.36](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.36.0).

@Koenkk can you update the [discussion](https://github.com/Koenkk/zigbee2mqtt/discussions/21462) to mention this (since the first post mentions dev branch, it appears to be causing confusion)?

**Reply by sam3487** [reply]
Thanks for clarifying! :)

**Reply by Koenkk** [reply]
Done!

**Comment by sam3487** [comment]
All of my devices have dissapeared from the frontend. They still appear to be working in the background though and are still listed in the devices yaml.

**Reply by sam3487** [reply]
`[2024-05-02 14:12:47] error: 	zh:ember: !!! NCP FATAL ERROR reason=ERROR_WRONG_DIRECTION. ATTEMPTING RESET... !!!
[2024-05-02 14:12:47] error: 	zh:ember:ezsp: ERROR Transaction failure; status=ERROR_WRONG_DIRECTION. Last: [FRAME: ID=52:"SEND_UNICAST" Seq=54 Len=26].
[2024-05-02 14:12:47] error: 	zh:controller:device: Default response to 0xf4b3b1fffeed1c6d failed
[2024-05-02 14:12:48] error: 	zh:ember:uart:ash: Received frame with CRC error
[2024-05-02 14:13:04] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:05] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:05] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:06] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:06] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:07] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:07] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:08] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:08] error: 	zh:ember: ~x~> [ZCL BROADCAST] Failed to send with status=NETWORK_BUSY.
[2024-05-02 14:13:09] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:09] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:10] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:10] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:11] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:11] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:12] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:12] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:13] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:15] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:15] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:16] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:16] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:17] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:17] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:13:18] error: 	zh:ember: [ZDO] Failed permit joining request with status=NETWORK_BUSY.
[2024-05-02 14:14:05] error: 	z2m: Failed to configure 'Guest Bathroom Bathtub Switch', attempt 1 (Error: ZCL command 0x00158d00041d1b17/11 `

**Reply by xelemorf** [reply]
Same for me, all devices dissapeared but working after a couple of hours. Tried to stop Z2M, delete database and coordinator backup file then reinstalling Z2M from scratch results the same. The UI also has some misplaced elements, permit join button under setting (not the dropdown) no longer visible. The dropdown permit join no longer works. Need to rollback it seems as it's completely dead at this point as cannot add any devices.

**Reply by Patricen1** [reply]
I do also have 49 devices in database.db files but only 45 appearing on frontend.
It looks like it was already the case with V1.36

**Comment by rwronski** [comment]
What does it mean when it's said that 'The ezsp driver is now deprecated'? Does it mean that it is no longer supported? Will my Sonoff dongle E still work with the EZSP driver?

**Reply by chris-1243** [reply]
Your Dongle-E is still supported. You will have to use `ember` instead of `ezsp` as a driver. You might need to update your coordinator firmware to min 7.4.x if it is still the case

**Reply by Grogger** [reply]
Do you happen to know if the Dongle-E's Multi-PAN RCP firmware is compatible?

**Reply by chris-1243** [reply]
@Grogger 

Check here #21462. That is the inital start of `ember`.
I guess @Nerivec might provide more explanations.

**Reply by Jonybat** [reply]
If switching driver and upgrading the coordinator firmware is required for 1.37.0, then I would say deprecated is not the best choice of words. It should read support removed or breaking change.

I spent a few minutes looking at the linked thread and couldn't find instructions or answer to this question either.

**Reply by rwronski** [reply]
I am currently using version 6.10.3 of this firmware: 
https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle-NoSigned/EZSP, and it works perfectly fine without any issues. Which version should I upgrade to that is well-tested and likely to ensure I encounter no problems?

**Reply by FuzzyMistborn** [reply]
I literally just updated my Sonoff -E's firmware using this site on Monday: https://darkxst.github.io/silabs-firmware-builder/ .  But the firmware version installed is 7.3.1.0 build 176.  Where can I get a 7.4.x firmware to install?

**Reply by N3rdix** [reply]
> I literally just updated my Sonoff -E's firmware using this site on Monday: https://darkxst.github.io/silabs-firmware-builder/ .  But the firmware version installed is 7.3.1.0 build 176.  Where can I get a 7.4.x firmware to install?

I used this file via custom url:
https://github.com/darkxst/silabs-firmware-builder/blob/main/firmware_builds/zbdonglee/ncp-uart-hw-v7.4.2.0-zbdonglee-115200.gbl

**Reply by Jonybat** [reply]
Can the firmware be updated without having to re-pair all the devices?

**Reply by Ra72xx** [reply]
I updated my SkyConnect several times now from the repo above without having to re-pair.

**Comment by jamesonuk** [comment]
Should I be expecting to see anything in logs in the front end?
Running in Docker and since moving to 1.37 and starting I have had zero log entries
![image](https://github.com/Koenkk/zigbee2mqtt/assets/1040621/1bc873bb-e2b1-41bb-813a-e3f45b94e140)
I see the release notes say debug logs are no longer in front end but normally there are lots of INFO logs I was still expecting to see...

In fact even looking inside the container the last entry was 30 minutes ago when I did the update.

Just to add restarting the container I see 
```
[2024-05-02 08:48:58] info:     z2m: Kitchen Cooker Window (0x30fb10fffed7c8a4): LED2005R5/LED2106R3 - IKEA TRADFRI bulb GU10, white spectrum, 345/380 lm (Router)
[2024-05-02 08:48:58] info:     z2m: Kitchen Island Window (0x60b647fffe5b2885): LED2005R5/LED2106R3 - IKEA TRADFRI bulb GU10, white spectrum, 345/380 lm (Router)
[2024-05-02 08:48:58] info:     z2m: Kitchen Island Door (0x60b647fffe8fa08f): LED2005R5/LED2106R3 - IKEA TRADFRI bulb GU10, white spectrum, 345/380 lm (Router)
[2024-05-02 08:48:58] info:     z2m: Kitchen Cooker Door (0x60b647fffe259b5e): LED2005R5/LED2106R3 - IKEA TRADFRI bulb GU10, white spectrum, 345/380 lm (Router)
[2024-05-02 08:48:58] info:     z2m: Sonoff Button (0x38398ffffeec7642): SNZB-01P - SONOFF Wireless button (EndDevice)
[2024-05-02 08:48:58] info:     z2m: Zigbee: disabling joining new devices.
[2024-05-02 08:48:59] info:     z2m: Connecting to MQTT server at mqtts://mosquitto.xxx:8883
[2024-05-02 08:48:59] info:     z2m: Connected to MQTT server
[2024-05-02 08:48:59] info:     z2m: Started frontend on port zigbee2mqtt.xxx:8080
[2024-05-02 08:48:59] info:     z2m: Zigbee2MQTT started!
```

but nothing else.

Is this expected now and we shouldn't expect to see things like (just some old logs I had around(
```
Info 2024-04-08 17:02:39Update of 'Spare Room TRV' at 100.00%
Info 2024-04-08 17:02:39MQTT publish: topic 'zigbee2mqtt/Spare Room TRV', payload '{"battery":100,"battery_low":null,"child_lock":"UNLOCK","closing_steps":414,"frost_protection_temperature":7,"idle_steps":38,"last_seen":"2024-04-08T16:02:36.409Z","linkquality":255,"local_temperature":19.6,"local_temperature_calibration":0,"occupied_heating_setpoint":16.5,"open_window":"OFF","running_state":"idle","system_mode":"heat","update":{"installed_version":4353,"latest_version":4357,"progress":100,"remaining":9,"state":"updating"},"update_available":false,"valve_closing_limit_voltage":2405,"valve_motor_running_voltage":1382,"valve_opening_limit_voltage":1260,"weekly_schedule":{"friday":"00:00/16 07:00/19 10:00/16 10:00/16 17:00/19 23:00/16","monday":"00:00/16 07:00/19 10:00/16 10:00/16 17:00/19 23:00/16","saturday":"00:00/16 07:00/19 10:00/19 10:00/19 10:00/19 23:00/16","sunday":"00:00/16 07:00/19 10:00/19 10:00/19 10:00/19 23:00/16","thursday":"00:00/16 07:00/19 10:00/16 10:00/16 17:00/19 23:00/16","tuesday":"00:00/16 07:00/19 10:00/16 10:00/16 17:00/19 23:00/16","wednesday":"00:00/16 07:00/19 10:00/16 10:00/16 17:00/19 23:00/16"}}'
Info 2024-04-08 17:02:39MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"Update of 'Spare Room TRV' at 100.00%","meta":{"device":"Spare Room TRV","progress":100,"status":"update_progress"},"type":"ota_update"}'
```

**Reply by Nerivec** [reply]
`z2m: Zigbee2MQTT started!` These (and before) would happen before frontend can register them.
`MQTT publish` These lines have been moved to `debug` since it was very spammy.
`Update of ` That would be OTA, so not frequent.

Depending on the adapter type you are using, it could be pretty quiet, unless some valuable `info` needs to be logged 😉
_If you want to make sure it is working, you can try Permit Join, that should add an `info` line._
_`info` level logging: https://github.com/search?q=org%3AKoenkk+logger.info+language%3ATypeScript&type=code&l=TypeScript_

**Reply by martyzeq** [reply]
I understand that it is a matter of "point of view", but this feels like info is almost useless to me, it should be more talkative imho - we have Info, Warn, Error and Debug - I am usually interested in state changes in log, and it was included in Info log level, which was just fine for me. Now I have to switch to Debug, which makes logs flooded. Would it be possible to extend loglevel setting to 8 values like severity levels in syslog ? I am really missing previous logging behaviour.... Thank you

**Reply by RCorsaletti** [reply]
i can't see devices messages even in debug, it was very useful at times to debug automations and stuff

**Comment by Ra72xx** [comment]
Are you certain that it is already prime time for the new `ember` driver? As much as I'm looking forward to it, as of now it is still unusable for me (#22249, while the old `ezsp` driver at least works most of the time.

**Comment by Milinator** [comment]
I can't rename devices anymore, the "edit" button is missing and the friendly name under settings tab is a read-only textfield.

**Reply by S33G** [reply]
Oh that's my bad.
Working on a fix right now.

**Reply by S33G** [reply]
https://github.com/nurikk/zigbee2mqtt-frontend/pull/2012

**Comment by KimpiegamesYT1** [comment]
I can no longer change the names of the devices?

**Reply by S33G** [reply]
Fixed here:
https://github.com/nurikk/zigbee2mqtt-frontend/pull/2012

**Comment by diesel437** [comment]
With this new update, will I have to repair all my devices if I change my coordinator hardware? Like I have a Sonoff Dongle-E dongle, and I would like to change to SMLight SLZB-06M. I'm avoiding this since there are 80 devices to re-pair.

**Reply by S33G** [reply]
No you should not need to re-pair providing you are using the same `.db`

**Reply by diesel437** [reply]
Ok, thats great. So docs need to be updated then: https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

**Reply by ministryofsillywalks** [reply]
Also would really like to know if this is true. 
My slzb-05 just died and now I would have to switch to a sonoff ZBDongle-E.
Can i make this switch without having to repair everything?

**Comment by N3rdix** [comment]
Just paired one Bosch device with install code to test the new ember, this basically went fine. But a new warning confuses me. Is this related to ember or is it the device telling that it's only compliant to an older revision and nothing to worry about?

`[2024-05-03 10:42:20] warning: 	zh:ember: [ZDO] Node descriptor for "35771" reports device is only compliant to revision "22" of the ZigBee specification (current revision: 23).`

EDIT: using a Sonoff Dongle-E with EZSP ncp-uart-hw-v7.4.2.0-zbdonglee-115200

**Reply by Nerivec** [reply]
It's just a warning when a device's firmware doesn't match current Zigbee revision; in case you ever have trouble with a device, that can sometimes be the cause. It should not cause problems as long as the versions aren't too far apart. If you see one with 'pre-21', that's when things usually get complicated.

**Reply by yamirWolf** [reply]
I have this problem, my device is on revision 21 and zigbee2mqtt in 23, how can I fix it?
Can you help me?

![IMG_7569](https://github.com/user-attachments/assets/1f38598b-bc89-447e-9080-2868abf81bdf)

**Reply by Geni0815** [reply]
I have exactly the same problem, does nobody have a solution?

**Reply by N3rdix** [reply]
unless you don't have any issues with the device I would ignore it (it's anyways just an info, no error). 
As far as i understand, there is no other way to make a device compatible with a newer revision than a firmware update which takes care of that...

**Reply by Geni0815** [reply]
but my Xiaomi LYWSD03MMC-z has not delivered any data to my Home Assistant since the battery change.
I can only switch the display on and off, as well as the comfort display.
Xiaomi has firmware version 3001-0123

**Reply by andye2801** [reply]
I have the same issue with a Phillips Hue dimmer switch, any solutions out there?

**Reply by nightfever77** [reply]
Same here with Phillips Hue dimmer switch, because action sensor is gone in HA

Fix: Enable these options in Z2M Settings / Home Assistant Integration:
![image](https://github.com/user-attachments/assets/8e32d022-bc81-4b51-987a-1132083c0c8e)

**Reply by arevinje** [reply]
Same here with my Namrom dimmer switches (listed as EcoDim Eco-Dim.07/Eco-Dim.10 in Z2M).

**Comment by merlinpimpim** [comment]
Hello,
There's something I don't understand about EZSP versus EMBER:
I use a SkyConnect coordinator in multiprotocol mode with this firmware (successfully for 3 months): https://github.com/darkxst/silabs-firmware-builder/blob/main/firmware_builds/skyconnect/rcp-uart-802154-v4.3.3-skyconnect-460800.gbl
It works fine with EZSP but not at all with EMBER, and I read that in the next release EZSP will no longer be supported.
What should I do?

**Reply by Koenkk** [reply]
>  and I read that in the next release EZSP will no longer be supported

This is not true, it will stay in but stop receiving maintenance. Still it is recommended to switch to `ember`.

**Reply by Nerivec** [reply]
You will definitely want to see https://youtu.be/8-YwXkgD3CY?t=2976 too. I'd advise you eventually switch out of multiprotocol, if you want to continue receiving timely updates.

**Reply by merlinpimpim** [reply]
Sorry Nerivec, I misinterpreted this information in the logs: "Deprecated driver 'ezsp' currently in use, 'ember' will become the officially supported EmberZNet driver in next release".

OK, I'll plan to switch back to Zigbee only.

**Comment by litecross91** [comment]
I got this in logs:
```

[2024-05-03 17:10:54] warning: 	zh:ember:ezsp: Received network/route error ROUTE_ERROR_INDIRECT_TRANSACTION_EXPIRY for "11906".
[2024-05-03 17:11:28] warning: 	zh:ember:ezsp: Received network/route error ROUTE_ERROR_NON_TREE_LINK_FAILURE for "11906".
```

What does it mean? It's a problem?

Zigbee2MQTT version
[1.37.0](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.37.0) commit: [unknown](https://github.com/Koenkk/zigbee2mqtt/commit/unknown)
Coordinator type
EmberZNet
Coordinator revision
7.4.2 [GA]
Coordinator IEEE Address
0xe0798dfffed1ba31
Frontend version
0.6.165
19.32.0
0.45.0

**Reply by HenryGlendening** [reply]
Did you find a way to address this? I get this error when I try to update a device

**Reply by hardwareadictos** [reply]
Same here

**Comment by litecross91** [comment]
I added several devices and the next line always came up since I set the "ember". 

`Node descriptor for "6402" reports device is only compliant to revision "21" of the ZigBee specification (current revision: 23).`
or
`zh:ember: [ZDO] Node descriptor for "37678" reports device is only compliant to revision "pre-21" of the ZigBee specification (current revision: 23).`
or
`zh:ember: [ZDO] Node descriptor for "31837" reports device is only compliant to revision "22" of the ZigBee specification (current revision: 23).`

Regardless, the devices work.

```
Info 2024-05-04 12:28:14Zigbee: allowing new devices to join.
Info 2024-05-04 12:28:14[STACK STATUS] Network opened.
Info 2024-05-04 12:28:54Interview for '0xa4c1381fa72ade21' started
Info 2024-05-04 12:28:54Device '0xa4c1381fa72ade21' joined
Info 2024-05-04 12:28:54Starting interview of '0xa4c1381fa72ade21'
Warning 2024-05-04 12:28:57[ZDO] Node descriptor for "6402" reports device is only compliant to revision "21" of the ZigBee specification (current revision: 23).
Info 2024-05-04 12:29:03Succesfully interviewed '0xa4c1381fa72ade21'
Info 2024-05-04 12:29:03Successfully interviewed '0xa4c1381fa72ade21', device has successfully been paired
Info 2024-05-04 12:29:03Device '0xa4c1381fa72ade21' is supported, identified as: Custom devices (DiY) Xiaomi temperature & humidity sensor with custom firmware (LYWSD03MMC)
Info 2024-05-04 12:29:03Configuring '0xa4c1381fa72ade21'
Info 2024-05-04 12:29:17Successfully configured '0xa4c1381fa72ade21'
```

**Reply by Nerivec** [reply]
See the Help section [here](https://nerivec.github.io/z2m-ember-helper/) for details on `ember`-specific logging. Some of this will be added to Z2M docs as appropriate, but in the meantime, I gathered it all there.

**Reply by litecross91** [reply]
I cannot find a similar error message

**Reply by Nerivec** [reply]
`Devices identified as having an older Zigbee revision. These can be the source of problems, especially if "pre-21".`
In short, this is just indicative, in case you ever encounter issues with that device, that could be the cause.

**Reply by litecross91** [reply]
Thank you!

**Comment by kvmgz** [comment]
Hi there. In this update my Symfonsik (sound) remote - I use it to turn on/off & dim lights - stopped working on button press, only the rotation is registered.
If I roll back to the previous version it works.
Any ideas?

**Comment by BenjaminShk** [comment]
Хочу написать мобильное приложение управляющее устройствами Zigbee через Z2M. Подскажите где найти документацию описывающую протокол общения с Z2M  или решение подобной задачи.

**Reply by BenjaminShk** [reply]
I want to write a mobile application that controls Zigbee devices via Z2M. Tell me where to find documentation describing the communication protocol with Z2M or a solution to a similar problem.

**Reply by N3rdix** [reply]
Maybe start at the docs related to MQTT topics: https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html

**Comment by Primevia** [comment]
My Bosch Thermostat ll devices cannot be full controlled anymore. I can only power it on but not power it off anymore. I can change the temperature up and down but that's all. I now have the devices powered on at 5 degrees even if it's summer. 
Anyone any idea if there will be a change to repair this issue. All my other devices are ok only the Bosch thermostat seems to have this issue. 
Thank you.

**Reply by N3rdix** [reply]
Do you have any errors or just missing attributes? Maybe it's related to recent refactorings to modernExtend...

**Reply by Primevia** [reply]
Hi N3rdix, i have no error Messages not in z2m nor on Home assistant. The three Bosch devices I have I cannot power off. Also if I try to run OTA there is no feedback while on other devices all works. Also I cannot rescan them. No messages back and it doesn't run a rescan. The thermostat card I have in Home assistant also only show heat but not off or automatic attribute as it was before.

**Comment by yamirWolf** [comment]
No body give a solution, i refund the device. I tried a lot of times the zigbee2mqtt-handler-conversion but nothing, i will try with other devives.

Un saludo
           Juan Carlos Pérez Navarro

On 1 Nov 2024, at 15:40, Geni0815 ***@***.***> wrote:

I have exactly the same problem, does nobody have a solution
